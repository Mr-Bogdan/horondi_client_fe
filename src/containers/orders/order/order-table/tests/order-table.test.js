import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as reactRedux from 'react-redux';

import OrderTable from '../order-table';

configure({ adapter: new Adapter() });

let wrapper;

function spyOnSelector(language, lightMode) {
  jest.spyOn(reactRedux, 'useSelector').mockImplementation((cb) =>
    cb({
      Language: { language },
      Theme: { lightMode }
    })
  );
}

const props = {
  currency: 0,
  calcPrice: 100,
  cartLoading: false,
  cartQuantityLoading: false,
  items: [{ product: { product_id: 'some id' }, price: 10 }],
  user: {}
};
describe('Order table component tests', () => {
  beforeAll(() => {
    jest.spyOn(reactRedux, 'useDispatch').mockImplementation(() => (cb) => cb());
  });

  it('should match snapshot', () => {
    spyOnSelector(0, false);

    wrapper = shallow(<OrderTable {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should cover other branches', () => {
    spyOnSelector(0, true);
    jest.spyOn(React, 'useState').mockImplementation(() => [true, () => null]);

    wrapper = shallow(<OrderTable {...props} />);
  });
});
