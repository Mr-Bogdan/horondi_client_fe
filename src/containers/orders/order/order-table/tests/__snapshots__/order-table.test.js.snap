// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order table component tests should match snapshot 1`] = `
<div
  className="makeStyles-root-3"
>
  <h2
    className="makeStyles-darkThemeTitleWrapper-2"
  >
    Ваш кошик
     
  </h2>
  <div
    className="makeStyles-table-4"
  >
    <WithStyles(ForwardRef(Table))>
      <WithStyles(ForwardRef(TableHead))>
        <WithStyles(ForwardRef(TableRow))
          classes={
            Object {
              "root": "makeStyles-darkThemeTableHeader-6",
            }
          }
        >
          <WithStyles(ForwardRef(TableCell))>
            Продукт
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))>
            Розмір
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))>
            Ціна
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))>
            Кількість
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))>
            До оплати
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))>
            Дія
          </WithStyles(ForwardRef(TableCell))>
        </WithStyles(ForwardRef(TableRow))>
      </WithStyles(ForwardRef(TableHead))>
      <WithStyles(ForwardRef(TableBody))>
        <CartItem
          calcPrice={100}
          cartLoading={false}
          cartQuantityLoading={false}
          currency={0}
          initialPrice={10}
          item={
            Object {
              "price": 10,
              "product": Object {
                "product_id": "some id",
              },
            }
          }
          key="some id"
          language={0}
          setModalItem={[Function]}
          setModalVisibility={[Function]}
          user={Object {}}
        />
      </WithStyles(ForwardRef(TableBody))>
    </WithStyles(ForwardRef(Table))>
  </div>
</div>
`;
